test/e2e/delete_test.go

```diff
@@ -270,3 +271,20 @@ func (tc *testContext) waitUntilDaemonsetScaled(name string, desiredReplicas int
 	}
 	return nil
 }
+
+// cleanupWorkloads attempts to delete all deployments that exist within the testContext workload namespace
+func (tc *testContext) cleanupDeployments() {
+	deployments, err := tc.client.K8s.AppsV1().Deployments(tc.workloadNamespace).List(context.TODO(), meta.ListOptions{})
```

Comment:
Do we need to wait for the deployments to be deleted? PS: I think I might have missed reviewing the master PR.